src/adapters/rethink.ts

```diff
@@ -100,10 +100,20 @@ export class RethinkAdapter extends AdapterBase {
         return rows[0];
     }
 
+    public async count<T>(
+        ctor: ModelCtor<T>,
+        predicate?: RethinkQueryPredicate<T>,
+    ): Promise<number> {
+        return await this.query(
+            ctor,
+            q => q.count(predicate),
+        );
+    }
+
     public query<T>(
         ctor: ModelCtor<T>,
         ...predicates: RethinkQueryPredicate<T[]>[]
-    ): rethinkdb.Term<T[]> {
+    ): rethinkdb.Term<any> {
```

Comment:
I feel this is kind of lazy. If we know what should be expected to return the count can't we set `T` to `<number>` on line 107? Instead of changing this to `any` as this breaks types when interfacing with the method.

Reply:
The problem here is that it will expect an Array of `T` as a return value. When requesting a count, Rethink will respond with a `number`, not an number Array. That makes the return value invalid.

Reply:
Let's not call people lazy, any is fine for now as its not really possible to tell whats going to be returned atm. This query method probably needs a rethink (no pun intended) later on.